fiaas_deploy_daemon/deployer/kubernetes/ingress.py

```diff
@@ -77,24 +80,45 @@ def _expand_default_hosts(self, app_spec):
         return [IngressItemSpec(host=host, pathmappings=all_pathmappings, annotations=None)
                 for host in self._generate_default_hosts(app_spec.name)]
 
-    def _group_ingresses_by_annotations(self, app_spec):
-        ''' Group the ingresses so that those with annotations are individual, keeping all without
-        annotations together
+    def _get_issuer_type(self, host):
+        for (suffix, issuer_type) in self._tls_issuer_type_overrides:
+            if host and (host == suffix or host.endswith("." + suffix)):
+                return issuer_type
+
+        return self._tls_issuer_type_default
+
+    def _group_ingresses(self, app_spec):
```

Comment:
Thinking out loud: I was wondering if the code would look simpler if we treat the certmanager annotations as regular annotations 🤔  This might mean an extra ingress for each overriden issuer though (i.e. counting towards rate limiting) and a refactor of the `apply` method to remove how the annotations are handled there. It wouldn't change how this feature behaves so probable for a later PR/refactor.